### DISCUSSION TITLE
For provideCompletionItems(): How can I detect if current position is inside a code block

### DISCUSSION THREAD

**Discussion by lymims** [microsoft/vscode-discussions#942] [Extension Development QnA] [answered]
I want to provide a completion for markdown files, but don't want it to trigger in code blocks. how do I detect it?

```
provideCompletionItems(document: vscode.TextDocument, position: vscode.Position) {
    const linePrefix = document.lineAt(position).text.substring(0, position.character);
    // detect if current position is inside a code block
    ...
}
```

**Comment by alefragnani** [comment, marked as answer]
Hi @lymims ,

The basic/built in Markdown language server (which powers the Markdown syntax highlighting) does not _expose_ the code blocks as separate symbols. You can confirm this, looking the **Outline View**, which only displays the _headings_. This means you can't rely on it to tell you if your cursor is located within code blocks. Otherwise, you could call `commands.executeCommand('vscode.executeDocumentSymbolProvider', window.activeTextEditor.document.uri)` to get the code blocks available in the Markdown file.

So, you will have to parse the Markdown file yourself, preferably/probably using some existing parser, in order to identify if the cursor is located within a code block. Or, you could use a simpler/naive approach, looking for the elements that encloses code blocks (single backticks, triple backticks/tildes, indent) both before and after the cursor position.

Hope this helps
